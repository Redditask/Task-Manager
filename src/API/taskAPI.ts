import {createAsyncThunk} from "@reduxjs/toolkit";

import {$authHost} from "./index";

import {ServerTask, Task} from "../types/types";

export const getTasks = createAsyncThunk<ServerTask [], number, {rejectValue: string}>(
  "tasks/getTasks",
    async function (userId, {rejectWithValue}){
        const response = await $authHost.get(`api/task/${userId}`);

        if (response.status !== 200){
            return rejectWithValue("Server error");
        }

        return response.data;
    }
);

export const postTask = createAsyncThunk<Task, {task: Task, userId: number}, {rejectValue: string}>(
    "tasks/postTask",
    async function ({task, userId}, {rejectWithValue, dispatch}) {
        task.id = new Date().toISOString();
        const response = await $authHost.post(`api/task/`, {task, userId});

        if (response.status !== 200) {
            return rejectWithValue("Server error");
        }

        return task;
    }
);

export const deleteTask = createAsyncThunk<{id: string}, {id: string, userId: number}, {rejectValue: string}>(
  "tasks/deleteTask",
  async function ({id, userId}, {rejectWithValue, dispatch}){
      const response = await $authHost.delete(`api/task/${userId}`, {data: {id}});

      if (response.status !== 200) {
          return rejectWithValue("Server error");
      }

      return {id};
  }
);

export const putTask = createAsyncThunk<Task, {task: Task, userId: number}, {rejectValue: string}>(
    "tasks, putTask",
    async function ({task, userId}, {rejectWithValue, dispatch}){
        const response = await $authHost.put(`api/task/${userId}`, {task});

        if (response.status !== 200) {
            return rejectWithValue("Server error");
        }

        return task
    }
);
