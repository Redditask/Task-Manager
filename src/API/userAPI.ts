import {$host} from "./index";
import jwtDecode from "jwt-decode";
import {User} from "../types/types";

export const registration = async (login: string, password: string): Promise<User> => {
    const {data} = await $host.post("api/user/registration", {login, password});
    return jwtDecode(data.token);
};

export const logIn = async (login: string, password: string): Promise<User> => {
    const {data} = await $host.post("api/user/login", {login, password});
    return jwtDecode(data.token);
};

export const auth = async (login: string, password: string) => {
    return await $host.post("api/user/auth", {login, password});
};
