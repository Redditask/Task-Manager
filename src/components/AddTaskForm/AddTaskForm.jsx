import styles from "./AddTaskForm.module.scss";

import React, {useState} from 'react';
import Input from "../UI/Input/Input";
import Button from "../UI/Button/Button";
import {useDispatch} from "react-redux";
import {addTask} from "../../store/taskManagerSlice";

const AddTaskForm = ({setModalStatus, date}) => {
    const [taskText, setTaskText] = useState("Enter your task");

    const today = new Date();
    let day=today.getDate();
    let month=today.getMonth();
    let year=today.getFullYear();

    if(date) [day, month, year] = date.split("-");


    const dispatch = useDispatch();
    const add = (text) => dispatch(addTask({
        taskText: text,
        year: Number(year),
        month: Number(month),
        day: Number(day)
    }))

    return (
        <div className={styles.AddTaskForm}>
            <h2 className={styles.AddTaskForm__title}>Enter a task</h2>
            <Input
                value={taskText}
                onChange={event=>setTaskText(event.target.value)}
            />
            <div className={styles.AddTaskForm__buttonArea}>
                {
                    taskText
                        ? <Button
                            text="Add task"
                            onClick={()=>{
                                add(taskText)
                                setModalStatus(false)
                            }}
                        />
                        : <Button
                            text="Input is empty"
                        />
                }
            </div>
        </div>
    );
};

export default AddTaskForm;
