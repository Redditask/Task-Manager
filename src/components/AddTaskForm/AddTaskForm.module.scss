.AddTaskForm {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  &__title{
    font-weight: normal;
    text-decoration: underline silver;

    color: black;

    margin: 1rem 0;
  }

  &__buttonArea{
    font-size: 1.5rem;

    margin-top: 1rem;
    margin-bottom: 2rem;

    border: 0.2rem silver solid;
    border-radius: 1rem;

    &:hover{
      background-color: silver;
      transition: 1s;
    }
  }
}
