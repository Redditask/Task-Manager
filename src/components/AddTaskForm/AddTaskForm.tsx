// @ts-ignore
import styles from "./AddTaskForm.module.scss";

import React, {useState} from 'react';

import {useAppDispatch} from "../../hooks/hooks";
import {addTask} from "../../store/taskManagerSlice";

import Input from "../UI/Input/Input";
import Button from "../UI/Button/Button";
import ColorPicker from "../ColorPicker/ColorPicker";
import TimePicker from "../TimePicker/TimePicker";

import {StringChangeEvent, Time} from "../../types/data";

interface AddTaskFormProps {
    setModalStatus: (modalStatus: boolean) => void;
    date: string;
}

const AddTaskForm:React.FC<AddTaskFormProps> = ({setModalStatus, date}) => {
    const [text, setText] = useState<string>("Your task");
    const [color, setColor] = useState<string>("beige");

    const [startTime, setStartTime] = useState<Time>({hour: 0, min: 0});
    const [endTime, setEndTime] = useState<Time>({hour: 23, min: 59});

    const today = new Date();
    let day: string = String(today.getDate());
    let month: string = String(today.getMonth());
    let year: string = String(today.getFullYear());

    if (date) [day, month, year] = date.split("-");

    const dispatch = useAppDispatch();

    const addTaskToStore = (text: string, color: string, startTime: Time, endTime: Time): void => {
        dispatch(addTask({
            taskText: text,
            year: Number(year),
            month: Number(month),
            day: Number(day),
            startTime: startTime,
            endTime: endTime,
            color: color,
        }));

        setModalStatus(false);
    };

    const submitAdd = (): void => addTaskToStore(text, color, startTime, endTime);

    const changeInputHandler = (event: StringChangeEvent): void => setText(event.target.value);

    return (
        <div className={styles.AddTaskForm}>
            <h2 className={styles.AddTaskForm__title}>Create a task</h2>
            <div className={styles.AddTaskForm__inputArea}>
                <Input
                    value={text}
                    onChange={changeInputHandler}
                />
                <TimePicker
                    startTime={startTime}
                    setStartTime={setStartTime}
                    endTime={endTime}
                    setEndTime={setEndTime}
                />
                <ColorPicker color={color} setColor={setColor}/>
            </div>
            <div className={styles.AddTaskForm__buttonArea}>
                {
                    text
                        ? <Button
                            title="Add task"
                            text="Create"
                            onClick={submitAdd}
                        />
                        : <Button
                            text="Input is empty"
                        />
                }
            </div>
        </div>
    );
};

export default AddTaskForm;
