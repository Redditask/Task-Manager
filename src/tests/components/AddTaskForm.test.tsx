import {fireEvent, render, screen} from "@testing-library/react";
import * as ReduxHooks from "react-redux";
import * as api from "../../API/taskAPI";

import AddTaskForm from "../../components/AddTaskForm/AddTaskForm";

jest.mock("react-redux");

const mockedUseDispatch = jest.spyOn(ReduxHooks, "useDispatch");
const mockedUseSelector = jest.spyOn(ReduxHooks, "useSelector");

describe("AddTaskForm", ()=>{

    it("should create AddTaskForm", ()=>{
        mockedUseSelector.mockReturnValueOnce(0);
        mockedUseDispatch.mockReturnValueOnce(jest.fn());

        const component = render(
            <AddTaskForm
                setModalStatus={jest.fn}
                date="17-5-2022"
            />
        );

        expect(component).toMatchSnapshot();
    });

    it("should dispatch thunk actions",()=>{
        mockedUseSelector.mockReturnValueOnce(2);
        const dispatch = jest.fn();
        mockedUseDispatch.mockReturnValueOnce(dispatch);

        const mockedPostTask = jest.spyOn(api, "postTask");

        render(
            <AddTaskForm
                setModalStatus={jest.fn}
                date="17-5-2022"
            />
        );

        fireEvent.click(screen.getByTitle("Add task"));

        expect(dispatch).toHaveBeenCalledTimes(1);
        expect(mockedPostTask).toHaveBeenCalledTimes(1);
        //standard input values
        expect(mockedPostTask).toHaveBeenCalledWith(
            {
                task: {
                    taskText: "Your task",
                    year: 2022,
                    month: 5,
                    day: 17,
                    startTime: {hour: 0, min: 0},
                    endTime: {hour: 23, min: 59},
                    color: "beige",
                },
                userId: 2
            }
        );
    });

    it("should properly change task data and post this",()=>{
        mockedUseSelector.mockReturnValue(2);
        const dispatch = jest.fn();
        mockedUseDispatch.mockReturnValue(dispatch);

        const mockedPostTask = jest.spyOn(api, "postTask");

        render(
            <AddTaskForm
                setModalStatus={jest.fn}
                date="17-5-2022"
            />
        );

        fireEvent.change(screen.getByTitle("Textarea"), {target: {value: "Added task text"}});
        fireEvent.change(screen.getByTitle("Start hour"), {target: {value: 20}});
        fireEvent.change(screen.getByTitle("Start minute"), {target: {value: 20}});
        fireEvent.change(screen.getByTitle("End minute"), {target: {value: 19}});
        fireEvent.change(screen.getByTitle("End hour"), {target: {value: 20}});
        fireEvent.change(screen.getByTitle("Task color"), {target: {value: "#4169E1"}});

        fireEvent.click(screen.getByTitle("Add task"));

        expect(dispatch).toHaveBeenCalledTimes(1);
        expect(mockedPostTask).toHaveBeenCalledTimes(1);
        expect(mockedPostTask).toHaveBeenCalledWith(
            {
                task: {
                    taskText: "Added task text",
                    year: 2022,
                    month: 5,
                    day: 17,
                    startTime: {hour: 20, min: 20},
                    endTime: {hour: 20, min: 21},
                    color: "#4169E1",
                },
                userId: 2
            }
        );
    });
});
