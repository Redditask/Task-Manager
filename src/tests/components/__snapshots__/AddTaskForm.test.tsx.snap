// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskForm should create AddTaskForm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="AddTaskForm"
      >
        <h2
          class="AddTaskForm__title"
        >
          Create a task
        </h2>
        <div
          class="AddTaskForm__inputArea"
        >
          <textarea
            class="Textarea"
            title="Textarea"
          >
            Your task
          </textarea>
          <div
            class="Container"
          >
            Start time:
            <div
              class="Container__startTime"
            >
              <select
                class="TimeSelect"
                title="Start hour"
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
                <option>
                  8
                </option>
                <option>
                  9
                </option>
                <option>
                  10
                </option>
                <option>
                  11
                </option>
                <option>
                  12
                </option>
                <option>
                  13
                </option>
                <option>
                  14
                </option>
                <option>
                  15
                </option>
                <option>
                  16
                </option>
                <option>
                  17
                </option>
                <option>
                  18
                </option>
                <option>
                  19
                </option>
                <option>
                  20
                </option>
                <option>
                  21
                </option>
                <option>
                  22
                </option>
                <option>
                  23
                </option>
              </select>
              <b
                style="font-size: 1.3rem;"
              >
                 : 
              </b>
              <select
                class="TimeSelect"
                title="Start minute"
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
                <option>
                  8
                </option>
                <option>
                  9
                </option>
                <option>
                  10
                </option>
                <option>
                  11
                </option>
                <option>
                  12
                </option>
                <option>
                  13
                </option>
                <option>
                  14
                </option>
                <option>
                  15
                </option>
                <option>
                  16
                </option>
                <option>
                  17
                </option>
                <option>
                  18
                </option>
                <option>
                  19
                </option>
                <option>
                  20
                </option>
                <option>
                  21
                </option>
                <option>
                  22
                </option>
                <option>
                  23
                </option>
                <option>
                  24
                </option>
                <option>
                  25
                </option>
                <option>
                  26
                </option>
                <option>
                  27
                </option>
                <option>
                  28
                </option>
                <option>
                  29
                </option>
                <option>
                  30
                </option>
                <option>
                  31
                </option>
                <option>
                  32
                </option>
                <option>
                  33
                </option>
                <option>
                  34
                </option>
                <option>
                  35
                </option>
                <option>
                  36
                </option>
                <option>
                  37
                </option>
                <option>
                  38
                </option>
                <option>
                  39
                </option>
                <option>
                  40
                </option>
                <option>
                  41
                </option>
                <option>
                  42
                </option>
                <option>
                  43
                </option>
                <option>
                  44
                </option>
                <option>
                  45
                </option>
                <option>
                  46
                </option>
                <option>
                  47
                </option>
                <option>
                  48
                </option>
                <option>
                  49
                </option>
                <option>
                  50
                </option>
                <option>
                  51
                </option>
                <option>
                  52
                </option>
                <option>
                  53
                </option>
                <option>
                  54
                </option>
                <option>
                  55
                </option>
                <option>
                  56
                </option>
                <option>
                  57
                </option>
                <option>
                  58
                </option>
                <option>
                  59
                </option>
              </select>
            </div>
            End time:
            <div
              class="Container__endTime"
            >
              <select
                class="TimeSelect"
                title="End hour"
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
                <option>
                  8
                </option>
                <option>
                  9
                </option>
                <option>
                  10
                </option>
                <option>
                  11
                </option>
                <option>
                  12
                </option>
                <option>
                  13
                </option>
                <option>
                  14
                </option>
                <option>
                  15
                </option>
                <option>
                  16
                </option>
                <option>
                  17
                </option>
                <option>
                  18
                </option>
                <option>
                  19
                </option>
                <option>
                  20
                </option>
                <option>
                  21
                </option>
                <option>
                  22
                </option>
                <option>
                  23
                </option>
              </select>
              <b
                style="font-size: 1.3rem;"
              >
                 : 
              </b>
              <select
                class="TimeSelect"
                title="End minute"
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
                <option>
                  5
                </option>
                <option>
                  6
                </option>
                <option>
                  7
                </option>
                <option>
                  8
                </option>
                <option>
                  9
                </option>
                <option>
                  10
                </option>
                <option>
                  11
                </option>
                <option>
                  12
                </option>
                <option>
                  13
                </option>
                <option>
                  14
                </option>
                <option>
                  15
                </option>
                <option>
                  16
                </option>
                <option>
                  17
                </option>
                <option>
                  18
                </option>
                <option>
                  19
                </option>
                <option>
                  20
                </option>
                <option>
                  21
                </option>
                <option>
                  22
                </option>
                <option>
                  23
                </option>
                <option>
                  24
                </option>
                <option>
                  25
                </option>
                <option>
                  26
                </option>
                <option>
                  27
                </option>
                <option>
                  28
                </option>
                <option>
                  29
                </option>
                <option>
                  30
                </option>
                <option>
                  31
                </option>
                <option>
                  32
                </option>
                <option>
                  33
                </option>
                <option>
                  34
                </option>
                <option>
                  35
                </option>
                <option>
                  36
                </option>
                <option>
                  37
                </option>
                <option>
                  38
                </option>
                <option>
                  39
                </option>
                <option>
                  40
                </option>
                <option>
                  41
                </option>
                <option>
                  42
                </option>
                <option>
                  43
                </option>
                <option>
                  44
                </option>
                <option>
                  45
                </option>
                <option>
                  46
                </option>
                <option>
                  47
                </option>
                <option>
                  48
                </option>
                <option>
                  49
                </option>
                <option>
                  50
                </option>
                <option>
                  51
                </option>
                <option>
                  52
                </option>
                <option>
                  53
                </option>
                <option>
                  54
                </option>
                <option>
                  55
                </option>
                <option>
                  56
                </option>
                <option>
                  57
                </option>
                <option>
                  58
                </option>
                <option>
                  59
                </option>
              </select>
            </div>
          </div>
          <p
            class="ColorPicker__title"
          >
            Theme:
          </p>
          <select
            class="ColorPicker"
            style="background-color: beige;"
            title="Task color"
          >
            <option
              selected=""
              style="background-color: beige;"
              value="beige"
            >
              beige
            </option>
            <option
              style="background-color: rgb(0, 255, 127);"
              value="#00FF7F"
            >
              green
            </option>
            <option
              style="background-color: rgba(205, 92, 92, 1);"
              value="#CD5C5CFF"
            >
              red
            </option>
            <option
              style="background-color: rgb(192, 192, 192);"
              value="#C0C0C0"
            >
              silver
            </option>
            <option
              style="background-color: rgb(65, 105, 225);"
              value="#4169E1"
            >
              blue
            </option>
          </select>
        </div>
        <div
          class="AddTaskForm__buttonArea"
        >
          <div
            class="Button"
            title="Add task"
          >
            Create
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="AddTaskForm"
    >
      <h2
        class="AddTaskForm__title"
      >
        Create a task
      </h2>
      <div
        class="AddTaskForm__inputArea"
      >
        <textarea
          class="Textarea"
          title="Textarea"
        >
          Your task
        </textarea>
        <div
          class="Container"
        >
          Start time:
          <div
            class="Container__startTime"
          >
            <select
              class="TimeSelect"
              title="Start hour"
            >
              <option>
                0
              </option>
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
              <option>
                16
              </option>
              <option>
                17
              </option>
              <option>
                18
              </option>
              <option>
                19
              </option>
              <option>
                20
              </option>
              <option>
                21
              </option>
              <option>
                22
              </option>
              <option>
                23
              </option>
            </select>
            <b
              style="font-size: 1.3rem;"
            >
               : 
            </b>
            <select
              class="TimeSelect"
              title="Start minute"
            >
              <option>
                0
              </option>
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
              <option>
                16
              </option>
              <option>
                17
              </option>
              <option>
                18
              </option>
              <option>
                19
              </option>
              <option>
                20
              </option>
              <option>
                21
              </option>
              <option>
                22
              </option>
              <option>
                23
              </option>
              <option>
                24
              </option>
              <option>
                25
              </option>
              <option>
                26
              </option>
              <option>
                27
              </option>
              <option>
                28
              </option>
              <option>
                29
              </option>
              <option>
                30
              </option>
              <option>
                31
              </option>
              <option>
                32
              </option>
              <option>
                33
              </option>
              <option>
                34
              </option>
              <option>
                35
              </option>
              <option>
                36
              </option>
              <option>
                37
              </option>
              <option>
                38
              </option>
              <option>
                39
              </option>
              <option>
                40
              </option>
              <option>
                41
              </option>
              <option>
                42
              </option>
              <option>
                43
              </option>
              <option>
                44
              </option>
              <option>
                45
              </option>
              <option>
                46
              </option>
              <option>
                47
              </option>
              <option>
                48
              </option>
              <option>
                49
              </option>
              <option>
                50
              </option>
              <option>
                51
              </option>
              <option>
                52
              </option>
              <option>
                53
              </option>
              <option>
                54
              </option>
              <option>
                55
              </option>
              <option>
                56
              </option>
              <option>
                57
              </option>
              <option>
                58
              </option>
              <option>
                59
              </option>
            </select>
          </div>
          End time:
          <div
            class="Container__endTime"
          >
            <select
              class="TimeSelect"
              title="End hour"
            >
              <option>
                0
              </option>
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
              <option>
                16
              </option>
              <option>
                17
              </option>
              <option>
                18
              </option>
              <option>
                19
              </option>
              <option>
                20
              </option>
              <option>
                21
              </option>
              <option>
                22
              </option>
              <option>
                23
              </option>
            </select>
            <b
              style="font-size: 1.3rem;"
            >
               : 
            </b>
            <select
              class="TimeSelect"
              title="End minute"
            >
              <option>
                0
              </option>
              <option>
                1
              </option>
              <option>
                2
              </option>
              <option>
                3
              </option>
              <option>
                4
              </option>
              <option>
                5
              </option>
              <option>
                6
              </option>
              <option>
                7
              </option>
              <option>
                8
              </option>
              <option>
                9
              </option>
              <option>
                10
              </option>
              <option>
                11
              </option>
              <option>
                12
              </option>
              <option>
                13
              </option>
              <option>
                14
              </option>
              <option>
                15
              </option>
              <option>
                16
              </option>
              <option>
                17
              </option>
              <option>
                18
              </option>
              <option>
                19
              </option>
              <option>
                20
              </option>
              <option>
                21
              </option>
              <option>
                22
              </option>
              <option>
                23
              </option>
              <option>
                24
              </option>
              <option>
                25
              </option>
              <option>
                26
              </option>
              <option>
                27
              </option>
              <option>
                28
              </option>
              <option>
                29
              </option>
              <option>
                30
              </option>
              <option>
                31
              </option>
              <option>
                32
              </option>
              <option>
                33
              </option>
              <option>
                34
              </option>
              <option>
                35
              </option>
              <option>
                36
              </option>
              <option>
                37
              </option>
              <option>
                38
              </option>
              <option>
                39
              </option>
              <option>
                40
              </option>
              <option>
                41
              </option>
              <option>
                42
              </option>
              <option>
                43
              </option>
              <option>
                44
              </option>
              <option>
                45
              </option>
              <option>
                46
              </option>
              <option>
                47
              </option>
              <option>
                48
              </option>
              <option>
                49
              </option>
              <option>
                50
              </option>
              <option>
                51
              </option>
              <option>
                52
              </option>
              <option>
                53
              </option>
              <option>
                54
              </option>
              <option>
                55
              </option>
              <option>
                56
              </option>
              <option>
                57
              </option>
              <option>
                58
              </option>
              <option>
                59
              </option>
            </select>
          </div>
        </div>
        <p
          class="ColorPicker__title"
        >
          Theme:
        </p>
        <select
          class="ColorPicker"
          style="background-color: beige;"
          title="Task color"
        >
          <option
            selected=""
            style="background-color: beige;"
            value="beige"
          >
            beige
          </option>
          <option
            style="background-color: rgb(0, 255, 127);"
            value="#00FF7F"
          >
            green
          </option>
          <option
            style="background-color: rgba(205, 92, 92, 1);"
            value="#CD5C5CFF"
          >
            red
          </option>
          <option
            style="background-color: rgb(192, 192, 192);"
            value="#C0C0C0"
          >
            silver
          </option>
          <option
            style="background-color: rgb(65, 105, 225);"
            value="#4169E1"
          >
            blue
          </option>
        </select>
      </div>
      <div
        class="AddTaskForm__buttonArea"
      >
        <div
          class="Button"
          title="Add task"
        >
          Create
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
